scipy/_build_utils/src/scipy_dll.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#if defined(SCIPY_DLL_EXPORTS)
+    #define SCIPY_DLL __declspec(dllexport)
+#elif defined(SCIPY_DLL_IMPORTS)
+    #define SCIPY_DLL __declspec(dllimport)
```

Comment:
```suggestion
// SCIPY_DLL
// inspired by https://github.com/abseil/abseil-cpp/blob/20240116.2/absl/base/config.h#L736-L753
//
// When building sf_error_state as a DLL, this macro expands to `__declspec(dllexport)`
// so we can annotate symbols appropriately as being exported. When used in
// headers consuming a DLL, this macro expands to `__declspec(dllimport)` so
// that consumers know the symbol is defined inside the DLL. In all other cases,
// the macro expands to nothing.
// Note: SCIPY_DLL_{EX,IM}PORTS are set in scipy/special/meson.build
#if defined(SCIPY_DLL_EXPORTS)
    #define SCIPY_DLL __declspec(dllexport)
#elif defined(SCIPY_DLL_IMPORTS)
    #define SCIPY_DLL __declspec(dllimport)
```

Reply:
There's a slight mismatch here, how the name `SCIPY_DLL` is scipy-wide, but it's defined in scipy/special. This isn't much of an issue right now, but if we ever have another shared library, and one shared library needs to use symbols from the other, then we'll need to split `SCIPY_DLL` into two, or rather, one per separate shared library.

Not a blocker for now, but noting it for context.

Reply:
That's a great point @h-vetinari! I did not at all consider the situation where we have two shared libraries and one of them needs symbols from the other. Like you said, there would be places  where we'd need separate macros, one expanding to  `__declspec(dllexport)` and the other to `__declspec(dllimport)`. I think I was just hoping to keep things simple for now.

Also, thanks for the comment! I was working on one locally, but was torn between documenting this specifically for `sf_error_state`, or giving the general conditions where one would use this. I'll consider this settled.